starport/pkg/cosmosanalysis/app/app.go

```diff
@@ -69,3 +73,165 @@ func CheckKeeper(path, keeperName string) error {
 	}
 	return nil
 }
+
+// FindRegisteredModules looks for all the registered modules in the App
+// It finds activated modules by checking if imported modules are registered in the app and also checking if their query clients are registered
+// It does so by:
+// 1. Mapping out all the imports and named imports
+// 2. Looking for the call to module.NewBasicManager and finds the modules registered there
+// 3. Looking for the implementation of RegisterAPIRoutes and find the modules that call their RegisterGRPCGatewayRoutes
+func FindRegisteredModules(pathToApp string) ([]string, error) {
+	fileSet := token.NewFileSet()
+	pkgs, err := parser.ParseDir(fileSet, pathToApp, nil, 0)
+	if err != nil {
+		return []string{}, err
+	}
+
+	var basicModules []string
+	for _, pkg := range pkgs {
+		for p, f := range pkg.Files {
+			if !strings.HasSuffix(p, appFileName) {
```

Comment:
Sounds reasonable! I will do that.
Quick question on the FindAppFilePath, should it take in the path to look for app.go in, or will it do a deep find or assume ./app/* when it searches?

Reply:
Yeah, that will work. I ended up sending in the chain root since we already have it and just going from "." doesn't work out when scaffolding the chain (and probably not when using --path either). But it seems to be working fine. 

Update coming soon.

Reply:
@ilgooz I updated the PR now with some changes to account for this. I also added some more tests cosmosanalysis since I was doing some refactoring to be able to reuse the logic already there. Let me know if it makes sense or you want any additional changes to it.

Reply:
That's great!